pyproject.toml

```
@@ -116,6 +117,10 @@ dev = [
     "anthropic",
     "google-generativeai;python_version>='3.9'",
     "langchain_core",
+    # Recent versions of langsmith (needed for langchain_core) cause a
+    # `argparse.ArgumentError` when running `pytest`.
+    # https://github.com/posit-dev/py-shiny/pull/1830
+    "langsmith<0.3",
```

Maybe link to the issue instead of the PR, since this PR doesn't provide any additional context.

```suggestion
    # langsmith (needed for langchain_core) versions >= 0.3
    # (up to at least 0.3.2 as of 2025-01-29) 
    # cause an `argparse.ArgumentError` when running `pytest`.
    # https://github.com/posit-dev/py-shiny/issue/1829
    "langsmith<0.3",
```